### PR TITLE
Garret/feature/santa integration

### PR DESCRIPTION
**Pull request by GarretReece**
Implements two tables that provide information about the state of santa on the host system.

### CODE REVIEW

File: santa/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2017 Trail of Bits, Inc.
```
Comment (mike-myers-tob) [inline santa/CMakeLists.txt]: 2018?

File: santa/src/santa.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+#include <string>
+
+#include <boost/iostreams/filter/gzip.hpp>
+#include <boost/iostreams/filtering_streambuf.hpp>
+
+#include <sqlite3.h>
+
+#include "santa.h"
+
+const std::string LOG_PATH = "/var/db/santa/santa.log";
```
Comment (mike-myers-tob) [inline santa/src/santa.cpp]: Are the Santa log and rules always at a fixed location or are they configurable? (just want to check that they are constant)

Comment (GarretReece) [reply santa/src/santa.cpp]: fixed location: https://github.com/google/santa/blob/d1d9762e29731c1857de25b40ebfe3b2b4b19bb0/Source/santad/SNTDatabaseController.m#L26

File: santa/src/santa.cpp [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                  bool save_to_archive = false) {
+  std::string line;
+  while (std::getline(incoming, line)) {
+    // explicitly filter to only include DENY events
+    if (line.find("decision=DENY") == std::string::npos) {
+      continue;
+    }
+
+    std::map<std::string, std::string> values;
+    extractValues(line, values);
```
[the remark this answers is not in the document]
Comment (GarretReece) [reply santa/src/santa.cpp]: It's legally okay to do so, in the sense that neither sqlite nor the map will complain about empty values.

Morally, I waffled over it for a bit, but ultimately decided that empty values are potentially better--the alternative is to pass in a list of required keys to be found, which seems burdensome here. I could see an argument for checking for the keys in the calling function, certainly.

File: santa/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    src/extension.h
+    src/extension.cpp
+
+    src/main.cpp
+  )
+
+  set(Boost_USE_STATIC_LIBS ON)
+  set(Boost_USE_MULTITHREADED ON)
+  set(Boost_USE_STATIC_RUNTIME ON)
+  find_package(Boost REQUIRED COMPONENTS iostreams)
```
Comment (alessandrogario) [inline santa/CMakeLists.txt]: We should document the required dependencies; do you think it is better to have it all written down in the root README or have a separate file in each extension folder?

Comment (alessandrogario) [reply santa/CMakeLists.txt]: Concerning Boost, I'll see if we can have the full lib inside the local repository (right now we only have the following components available: filesystem, context, thread, chrono, regex, coroutine, system)

**Pull request merged**